src/main/java/microsoft/sql/Vector.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package microsoft.sql;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public final class Vector implements java.io.Serializable {
+    public enum VectorDimensionType {
+        F16, // 16-bit (half precision) float
+        F32 // 32-bit (single precision) float
+    }
+
+    public VectorDimensionType vectorType;
+    public int dimensionCount;
+
+    public float[] data;
+
+    public Vector(int dimensionCount, VectorDimensionType vectorType, float[] data) {
+        this.dimensionCount = dimensionCount;
+        this.vectorType = vectorType;
+        this.data = data;
+    }
+
+    /**
+     * Converts a byte array to a Vector object. The byte array must contain the following:
+     * 8 bytes for header and 4 bytes per float value.
+     */
+    public static microsoft.sql.Vector fromBytes(byte[] bytes) {
+        if (bytes == null) {
+            return null;
+        }
+        if (bytes.length < 8) {
+            throw new IllegalArgumentException("Byte array length must be at least 8 bytes.");
+        }
+        if (bytes.length % 4 != 0) {
+            throw new IllegalArgumentException("Byte array length must be a multiple of 4.");

```

Comment:
Can we add the below error message in SQLServerResource.java
{"R_InvalidVectorLength", "Byte array length must be at least \"{0}\" bytes."}